AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -415,10 +422,13 @@ final public class AWSMobileClient: _AWSMobileClient {
             
             let federationProviderIdentifier = hostedUIOptions.federationProviderName
             
-            var scopes = infoDictionary?["Scopes"] as? [String]
-            
             if hostedUIOptions.scopes != nil {
-                scopes = hostedUIOptions.scopes
+                self.scopes = hostedUIOptions.scopes
```

Comment:
self.scopes = hostedUIOptions.scopes ?? infoDictionary?["Scopes"] as? [String]

Reply:
we also have to set a local boolean variable to indicate that custom scopes are specified using that conditional

Reply:
Done

Reply:
Done